packages/main/cypress/specs/TableNavigation.cy.tsx

```diff
@@ -0,0 +1,326 @@
+		cy.get("@headerRow").should("be.focused")
+
+			.type("{uparrow}");
+		cy.get("@headerRow").should("be.focused")
+
+			.type("{downarrow}{downarrow}");
+		cy.get("@rows").eq(1).should("be.focused")
+
+			.type("{pagedown}");
+		cy.get("@rows").eq(21).should("be.focused")
+
+			.type("{pagedown}");
+		cy.get("@rows").eq(24).should("be.focused")
+
+			.type("{pagedown}");
+		cy.get("#growing").shadow().find("#growing-button").should("be.focused")
+
+			.type("{pageup}");
+		cy.get("@rows").eq(5).should("be.focused")
+
+			.type("{pageup}");
+		cy.get("@rows").eq(0).should("be.focused")
+
+			.type("{end}");
+		cy.get("@rows").eq(24).should("be.focused")
+
+			.type("{end}");
+		cy.get("#growing").shadow().find("#growing-button").should("be.focused")
+
+			.type("{home}");
+		cy.get("@rows").eq(0).should("be.focused")
+
+			.type("{home}");
+		cy.get("@headerRow").should("be.focused");
+	});
+
+	it("should navigate on cells", () => {
+		cy.get("@rows").eq(0)
+			.click("left");
+
+		cy.get("@rows").eq(0).should("be.focused")
+
+			.type("{rightarrow}");
+		getCell(0, 0, false).should("be.focused")
+
+			.type("{leftarrow}");
+		cy.get("@rows").eq(0).should("be.focused")
+
+			.type("{rightarrow}{rightarrow}");
+		getCell(0, 1, false).should("be.focused")
+
+			.type("{home}");
+		getCell(0, 0, false).should("be.focused")
+
+			.type("{end}");
+		getCell(0, 3, false).should("be.focused")
+
+			.type("{rightarrow}");
+		getCell(0, 3, false).should("be.focused")
+
+			.type("{end}");
+		cy.get("@rows").eq(0).should("be.focused")
+
+			.type("{end}");
+		cy.get("@rows").eq(24).should("be.focused")
+
+			.type("{rightarrow}{rightarrow}{rightarrow}");
+		getCell(24, 2, false).should("be.focused")
+
+			.type("{pageup}");
+		getCell(4, 0, false).should("be.focused")
+
+			.type("{pageup}");
+		getCell(0, 0, false).should("be.focused")
+
+			.type("{pageup}");
+		getCell(0, 0, true).should("be.focused")
+
+			.type("{pagedown}");
+		getCell(19, 0, false).should("be.focused")
+
+			.type("{pagedown}");
+		getCell(24, 0, false).should("be.focused")
+
+			.type("{pagedown}");
+		cy.get("#growing").shadow().find("#growing-button").should("be.focused")
+
+			.type("{home}");
+		cy.get("@rows").eq(0).should("be.focused")
+
+			.type("{home}");
+		cy.get("@headerRow").should("be.focused");
+
+		cy.get("@headerRow").type("{downarrow}{rightarrow}");
+
+		for (let i = 0; i < 3; i++) {
+			getCell(i, i, false).should("be.focused")
+				.type("{downarrow}{rightarrow}");
+		}
+
+		getCell(3, 3, false).should("be.focused")
+			.type("{downarrow}");
+
+		getCell(4, 0, false).should("be.focused");
+	});
+
+	it("should handle F2/F7/Enter/Tab/Up/Down", () => {
+		cy.get("@rows").eq(0).get("#row1-input").as("row1Input");
+		cy.get("@rows").eq(1).get("#row2-input").as("row2Input");
+
+		cy.get("@rows").eq(0).click("left");
+		cy.get("@rows").eq(0).should("be.focused");
+
+		cy.get("@rows").eq(0).realPress("F2");
+		cy.get("@row1Input").eq(0).should("be.focused")
+
```

Comment:
Again. There are multiple instances in the code, where formatting is just weird. please have a look again